statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,508 @@
+
+                # If there is only one or two clusters then weights wont change
+                if (self.design.ncs[s] == 1 or self.design.ncs[s] == 2):
+                    continue
+                # array of clusters to resample from
+                ii = np.flatnonzero(self.design.sfclust == s)
+                # resample them
+                ii_resample = np.random.choice(ii, size=(self.design.ncs[s]-1))
+                # accumulate number of times cluster i was resampled
+                bin += np.bincount(ii_resample,
+                                   minlength=max(self.design.sclust)+1)
+            # augment weights
+            w *= bin[self.design.sclust]
+            # call the stat w/ the new weights
+            jdata.append(stat._stat(weights=w))
+        jdata = np.asarray(jdata)
+        # nh = self.design.ncs[self.design.sfclust].astype(np.float64)
+        # pseudo = jdata + nh[:, None] * (np.dot(w, stat.data) - jdata)
+
+        boot_mean = jdata.mean(0)
+        var = ((jdata - boot_mean)**2).sum(0) / (replicates - 1)
+        est = stat._stat(self.design.weights)
+        return est, var
+
+    def jack(self, stat):
+        """
+        Jackknife variance estimation for survey data.
+
+        Parameters
+        ----------
+        stat : object
+            Object of class SurveyMean, SurveyTotal, SurveyPercentile, etc
+
+        Returns
+        -------
+        est : ndarray
+            The point estimates of the statistic, calculated on the columns
+            of data.
+        vc : square ndarray
+            The variance-covariance matrix of the estimates, obtained using
+            the (drop 1) jackknife procedure.
+        pseudo : ndarray
+            The jackknife pseudo-values.
+        """
+
+        jdata = []
+        est = []
+        # for each cluster
+        for c in range(self.design.nclust):
+            # get stratum that the cluster belongs in
+            s = self.design.sfclust[c]
+            nh = self.design.ncs[s]
+            self.w = self.design.weights.copy()
+            # all weights within the stratum are modified
+            self.w[self.design.strat == s] *= nh / float(nh - 1)
+            # but if you're within the cluster to be removed, set as 0
+            self.w[self.design.sclust == c] = 0
+            # if dealing w/ survey quantile...
+            if hasattr(stat, "quantile"):
+                # 3d array, nclust x col x len(quantiles)
+                jdata.append([stat._stat(self.w, index) for
+                              index in range(stat.data.shape[1])])
+                est.append([stat._stat(self.design.weights, index) for
+                            index in range(stat.data.shape[1])])
+            else:
+                jdata.append(stat._stat(self.w))
+        jdata = np.asarray(jdata)
+
+        nh = self.design.ncs[self.design.sfclust].astype(np.float64)
+        # pseudo = jdata + nh[:, None] * (np.dot(self.w, stat.data) - jdata)
```

Comment:
np.dot(self.w, stat.data) is an array w/ shape (num cols, ) but jdata for quantiles has shape (num clusters, num of percentiles requested). in this case I'm subtracting something w/ shape (2, ) and (5, 6) so just commented it out for now to make sure everything else works